src/views/bridge-details/bridge-details.view.tsx

```diff
@@ -133,7 +133,19 @@ const BridgeDetails: FC = () => {
           depositCount,
           networkId,
         })
-          .then((bridge) => {
+          .then(async (bridge) => {
+            const tx = await connectedProvider.data.provider.getTransaction(bridge.depositTxHash);
+
+            if (!tx) {
+              throw new Error(`DepositTx not found: ${bridge.depositTxHash}`);
+            }
+
+            const allowedAccounts = [tx.from, bridge.destinationAddress];
+
+            if (!allowedAccounts.includes(connectedProvider.data.account)) {
+              return navigate(routes.activity.path);
+            }
```

Comment:
Why allow `tx.from` considering that the API of the bridge only returns the deposits belonging to the `destinationAddress`?

In other words, if I have the addresses `0xA` and `0xB` and I bridge some funds from `0xA` specifying `0xB` as `destinationAddress`, while the selected account in MetaMask is `0xA`, a request `/bridges/0xA` won't return this deposit.

If I need to change the account to `0xB` to see the deposit in the list, I think we should be consistent and hide the deposit from `0xA`.

The reasoning is based on the assumption that as a user (account) I should not see the details of a deposit which does not belong to me (i.e, is not in the list for the selected account).


```diff
diff --git a/src/views/bridge-details/bridge-details.view.tsx b/src/views/bridge-details/bridge-details.view.tsx
index aab33a2..363ff43 100644
--- a/src/views/bridge-details/bridge-details.view.tsx
+++ b/src/views/bridge-details/bridge-details.view.tsx
@@ -140,9 +140,7 @@ const BridgeDetails: FC = () => {
               throw new Error(`DepositTx not found: ${bridge.depositTxHash}`);
             }

-            const allowedAccounts = [tx.from, bridge.destinationAddress];
-
-            if (!allowedAccounts.includes(connectedProvider.data.account)) {
+            if (connectedProvider.data.account !== bridge.destinationAddress) {
               return navigate(routes.activity.path);
             }
```

Reply:
Mmmm you are right! I didn't remember that the API is returning only the deposits for the `destinationAddress`, so it doesn't make sense to take `tx.from` into account :'') Fixed.